### DISCUSSION TITLE
Null check operator used on a null value

### DISCUSSION THREAD

**Discussion by goflutterjava** [ethanblake4/dart_eval#23] [General]
original class
```
class OpenApiAsyncDebug {
  final name = "open async debug";

  Future<String> helloWorld() async {
    return "helloWorld";
  }

  Future<String> echo(String str) async {
    return str;
  }
}
```
bridge class
```
import 'package:dart_eval/dart_eval_bridge.dart';

class OpenApiAsyncBridge {
  static const $type = BridgeTypeReference.unresolved(
      BridgeUnresolvedTypeReference(
          'package:paas_dashboard_flutter/open/open_api_debug_async.dart', 'OpenApiAsyncDebug'),
      []);

  static const $declaration = BridgeClassDeclaration($type, isAbstract: false, constructors: {}, methods: {
    'helloWorld': BridgeMethodDeclaration(
        false,
        BridgeFunctionDescriptor(
            BridgeTypeAnnotation(
                BridgeTypeReference.unresolved(BridgeUnresolvedTypeReference('dart:core', 'Future'), []), false),
            {},
            [],
            {})),
  }, getters: {}, setters: {}, fields: {});
}
```
eval code
```
static void evalPiece(String piece) {
    var compiler = Compiler();
    compiler.defineBridgeClasses([OpenApiAsyncBridge.$declaration]);
    final code = "void main() async {\n"
        + "final openApiSyncDebug = OpenApiSyncDebug();\n"
        + "final openApiAsyncDebug = OpenApiAsyncDebug();\n"
        + piece + "\n"
        + "}";
    final program = compiler.compile({
      'package:example': {
        'main.dart': code
      }
    });
    final runtime = Runtime.ofProgram(program);
    runtime.execute(0);
  }
```
Error Message
```
Failed to load "/Users/xxxxxx/code/flutter/paas_dashboard_flutter/test/eval/eval_service_test_debug_async_helloworld.dart": Null check operator used on a null value
```

**Comment by ethanblake4** [comment]
There are a lot of problems. You are trying to create an `OpenApiSyncDebug()` but you did not make a bridge declaration for that, you only made it for the `OpenApiAsyncDebug`. In addition you did not import the library (`package:paas_dashboard_flutter/open/open_api_debug_async.dart`) in the eval code. Even for the async one you only made the compiler bridge `$declaration` but it's not a runtime bridge, the OpenApiAsyncBridge needs to `extends OpenApiAsyncDebug with $Bridge<OpenApiAsyncDebug>` and override the $bridgeGet, $bridgeSet, helloWorld, echo, etc. Also the compiler bridge `$declaration` needs to have the default constructor, and you need to register the bridge constructor at runtime with `runtime.registerBridgeFunc`. Finally you should not use `runtime.execute(0)` but instead `runtime.executeNamed(0, 'main')`.

I would suggest reading the `dart_eval_example.dart` more carefully, all of these things are correctly done in there.

**Comment by ethanblake4** [comment]
Converted this to a discussion as it is not a bug or feature request.
